test_ffi/src/lib.rs

```diff
@@ -6,6 +8,18 @@ pub extern "C" fn print_something() {
   println!("something");
 }
 
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+#[no_mangle]
+pub extern "C" fn print_buffer(ptr: *const u8, len: usize) {
+  let buf = unsafe { std::slice::from_raw_parts(ptr, len) };
+  println!("{:?}", buf);
+}
+
+#[no_mangle]
+pub extern "C" fn return_buffer() -> *const u8 {
+  [1, 2, 3, 4, 5, 6, 7, 8].as_ptr()
+}
```

Comment:
Discussed offline with @piscisaureus and we're gonna remove buffer-return-values for now and only keep buffer-as-arg